### ISSUE TITLE
Include for "Arduino.h" inserted too late

### ISSUE THREAD

**Issue by nickgammon** [arduino/Arduino#1518]
Example code ...

Sketch:

```
#include "test.h"

void setup () { }
void loop () { }
```

File "test.h":

```
byte x;
```

Error:

```
In file included from sketch_jul24c.ino:1:
test.h:1: error: 'byte' does not name a type
```

Generated code from pre-processor:

```
#line 1 "sketch_jul24c.ino"
#include "test.h"

#include "Arduino.h"
void setup ();
void loop ();
#line 3
void setup () { }
void loop () { }
```

If the include for "Arduino.h" had been done first (before the rest of the sketch was copied in) then the declarations in test.h would work. As it is, you now have to add an include into the front of all the .h files which defeats the usefulness of having the automatic inclusion in the main sketch.  

IDE 1.0.5.

These look related: 

https://github.com/arduino/Arduino/issues/1310
https://github.com/arduino/Arduino/issues/1328
https://github.com/arduino/Arduino/issues/1468

**Comment by nickgammon** [comment]
Similar problem. In IDE 1.6.4:

```
#ifdef __AVR_ATmega2560__
  const byte CLOCKOUT = 11;  // Mega 2560
#else
  const byte CLOCKOUT = 9;   // Uno, Duemilanove, etc.
#endif

void setup () { }
void loop () { }
```

Error:

```
sketch_jun27a:8: error: 'byte' does not name a type
'byte' does not name a type
```

**Comment by ffissore** [comment]
Fixed by #3779
